client/login/controller.js

```diff
@@ -156,10 +157,14 @@ export function redirectDefaultLocale( context, next ) {
 		return next();
 	}
 
+	// Add querystring to redirect only comming from Partner Signup
+	const redirectQueryString = isPartnerSignupQuery( context.query )
+		? '?' + context.querystring
+		: '';
 	if ( context.params.isJetpack === 'jetpack' ) {
 		page.redirect( '/log-in/jetpack' );
 	} else {
-		page.redirect( '/log-in' );
+		page.redirect( '/log-in' + redirectQueryString );
```

Comment:
This looks like a pre-existing bug, why do we want to drop the query param here?

On the login screen the url to the english version looks correct:

https://wordpress.com/log-in/en?client_id=50916&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fresponse_type%3Dcode%26client_id%3D50916%26state%3Dsnip%26redirect_uri%3Dhttps%253A%252F%252Fwoocommerce.com%252Fwc-api%252Fwpcom-signin%253Fnext%253Dpartner-signup%26blog_id%3D0%26wpcom_connect%3D1%26wccom-from%26calypso_env%3Dproduction%26redirect_to%3Dhttps%3A%2F%2Fwoocommerce.com%2Fpartner-signup%2F

We then drop all of that and redirect it away? Why would we want that?

Line 137 mentions we only want to handle simple routes, what happens when we add a check to that if statement to ensure the query string is empty?

Reply:
I saw the same here, all this function is unnecessary or is there for a reason, so the best way to "not mess it" was to do this `hack`

Reply:
We might just need to do this in the end but can we at least make sure we understand the code properly before adding to it?

e.g.

- Why does this function exists (what's its purpose)?
- Why do we need to redirect here at all? (what use cases touch this code path)
- Why do we want to drop the querystring for these redirects?

Reply:
Code was [introduced here](https://github.com/Automattic/wp-calypso/pull/19319/files?w=1#diff-c8d0eae83cb51d69c9501a45dcc73981b2ba3ab149bb63b0d4b3dddf841f6545R110-R131).

There's a recent change [here](https://github.com/Automattic/wp-calypso/pull/63271) (reverted), reattempted/merged [here](https://github.com/Automattic/wp-calypso/pull/63660).

I spent a bit of time today looking at if we can remove the whole function but I'm not totally sure I understand it's full purpose yet or how our state hydration is working. The comment to this link in the code was somewhat helpful: pd2qbF-P-p2#comment-20